renderer.go

```diff
@@ -101,11 +108,19 @@ type Renderer struct {
 	ChromaOptions []html.Option
 	Style         *chroma.Style
 	Formatter     *html.Formatter
+	embedCSS bool
 }
 
 // RenderNode satisfies the Renderer interface
 func (r *Renderer) RenderNode(w io.Writer, node *bf.Node, entering bool) bf.WalkStatus {
 	switch node.Type {
+	case bf.Document:
+		if entering && r.embedCSS {
+			w.Write([]byte("<style>"))
+			r.Formatter.WriteCSS(w, r.Style)
+			w.Write([]byte("</style>"))
```

Comment:
The errors aren't handled here, so maybe we could move that into another function that would return an error if something doesn't work, and fallback to `r.Base.RenderNode` if an error presents itself.
Something like `(r *Renderer) renderCSS(w io.Writer) error` which would first try to write the CSS to an internal buffer before actually writing the style tags, as to not polute the document with unnecessary tags 
What do you think of that?

Reply:
The thing is if you don't write CSS the document is screwed either way -  doing rest of the formatting but without required CSS still makes a mess out of page. 

Also half-written style within <style> boundaries should be pretty safe either way.

I'd also rather not add random buffers for basically static data for no good reason (I'm actually not sure why Chroma styles aren't just rendered once then cached)

But you're right, it *would* be nice is signalling that error to ease debugging but I'm not sure what's the way to do it with blackfriday.

Reply:
Yes, I took a look at Blackfriday v2 and there's quite no way to return an error, or at least it's not obvious. You're also right about the screwed up document. So that's not really an issue and indeed it would add a random buffer, which isn't really useful.

Reply:
I was thinking about just writing `/* chroma css error: */` if it happens but that's probably a bad idea...